rom/dev/src/flow/cold_reset/mod.rs

```diff
@@ -73,18 +81,33 @@ impl ColdResetFlow {
     }
 }
 
+// Writes out a u32 length in little-endian byte order, followed by the TBS.
 pub fn copy_tbs(tbs: &[u8], tbs_type: TbsType) -> CaliptraResult<()> {
-    let dst = match tbs_type {
-        TbsType::LdevidTbs => unsafe {
-            let ptr = &mut LDEVID_TBS_ORG as *mut u8;
-            core::slice::from_raw_parts_mut(ptr, tbs.len())
-        },
-        TbsType::FmcaliasTbs => unsafe {
-            let ptr = &mut FMCALIAS_TBS_ORG as *mut u8;
-            core::slice::from_raw_parts_mut(ptr, tbs.len())
-        },
+    const SIZE_LEN: usize = core::mem::size_of::<u32>();
+    const MAX_TBS_LEN: usize = 0x400 - SIZE_LEN;
+
+    if tbs.len() > MAX_TBS_LEN {
+        raise_err!(TbsUnsupportedDataLength);
+    }
+
+    let (len_dst, tbs_dst) = unsafe {
+        let ptr = match tbs_type {
+            TbsType::LdevidTbs => &mut LDEVID_TBS_ORG,
+            TbsType::FmcaliasTbs => &mut FMCALIAS_TBS_ORG,
+        } as *mut u8;
+
+        ptr.write_bytes(0, SIZE_LEN);
```

Comment:
I somewhat prefer using the hand-off table instead of adding structure to the cert portions of DCCM. It's easier to keep track of structure changes between different firmware stages if they're in the same place.